renderer/components/action-bar/record-button.js

```diff
@@ -1,39 +1,48 @@
+import electron from 'electron';
 import React from 'react';
 import PropTypes from 'prop-types';
 
 class RecordButton extends React.Component {
   state = {}
 
   componentDidMount() {
-    navigator.getUserMedia({
-      audio: true
-    }, stream => {
-      const audioContext = new AudioContext();
-      const analyser = audioContext.createAnalyser();
-      const microphone = audioContext.createMediaStreamSource(stream);
-      const javascriptNode = audioContext.createScriptProcessor(2048, 1, 1);
-
-      analyser.smoothingTimeConstant = 0.8;
-      analyser.fftSize = 1024;
-
-      microphone.connect(analyser);
-      analyser.connect(javascriptNode);
-
-      javascriptNode.connect(audioContext.destination);
-      javascriptNode.onaudioprocess = () => {
-        const array = new Uint8Array(analyser.frequencyBinCount);
-        analyser.getByteFrequencyData(array);
-
-        const avg = array.reduce((p, c) => p + c) / array.length;
-        if (avg >= 36) {
-          this.setState({showFirst: true, showSecond: true, shouldStop: false});
-        } else {
-          this.setState({shouldStop: true});
+    const settings = electron.remote.require('./common/settings');
+    const recordAudio = settings.get('recordAudio');
+    const audioInputDeviceId = settings.get('audioInputDeviceId');
+
+    if (recordAudio) {
+      navigator.getUserMedia({
+        audio: {
+          deviceId: audioInputDeviceId
         }
-      };
-    }, err => {
-      console.err('The following error occured: ' + err.name);
-    });
+      }, stream => {
+        const audioContext = new AudioContext();
+        const analyser = audioContext.createAnalyser();
+        const microphone = audioContext.createMediaStreamSource(stream);
+        const javascriptNode = audioContext.createScriptProcessor(2048, 1, 1);
+
+        analyser.smoothingTimeConstant = 0.8;
+        analyser.fftSize = 1024;
+
+        microphone.connect(analyser);
+        analyser.connect(javascriptNode);
+
+        javascriptNode.connect(audioContext.destination);
+        javascriptNode.onaudioprocess = () => {
+          const array = new Uint8Array(analyser.frequencyBinCount);
+          analyser.getByteFrequencyData(array);
+
+          const avg = array.reduce((p, c) => p + c) / array.length;
+          if (avg >= 36) {
+            this.setState({showFirst: true, showSecond: true, shouldStop: false});
+          } else {
+            this.setState({shouldStop: true});
+          }
+        };
+      }, err => {
+        console.error('The following error occured: ' + err.name);
```

Comment:
This should be reported to Sentry too.

And would be better to do:

```js
console.error('An error occurred when trying to get audio levels:', err);
```

Reply:
I haven't hooked up Sentry yet, but when that's in, absolutely 👍